boa/src/builtins/map/mod.rs

```diff
@@ -333,12 +357,23 @@ impl Map {
     ///
     /// [spec]: https://tc39.es/ecma262/#sec-map.prototype.clear
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/clear
-    pub(crate) fn clear(this: &JsValue, _: &[JsValue], _: &mut Context) -> JsResult<JsValue> {
-        this.set_data(ObjectData::map(OrderedMap::new()));
-
-        Self::set_size(this, 0);
```

Comment:
Missed the `set_size` on the new version. That's why some tests now fail. You can put if after the `if let`, but we should really consider implementing the `get Map.prototype.size` accessor instead of managing the state ourselves.